### ISSUE TITLE
Move parameter ports to top of the box

### ISSUE THREAD

**Issue by bergie** [noflo/noflo-ui#469]
We have two types of input ports: *inputs* that are used in processing, and *parameters* that configure the behavior of the component but don't actually cause it to execute. For an example, see how [NoFlo's WirePattern is configured](https://github.com/noflo/noflo-mqtt/blob/master/components/SendMessage.coffee#L38)

Currently both types are rendered to the left of the node in the graph:

![screenshot 2015-04-02 at 17 20 52](https://cloud.githubusercontent.com/assets/3346/6966995/e0751e38-d95c-11e4-9049-8a40ec07c762.png)

Moving the params to the top of the box, like in [IDEF0](http://en.wikipedia.org/wiki/IDEF0#IDEF0_Building_blocks) would help to clarify the differences between these port types.

**Comment by jonnor** [comment]
Would need to define "parameters" versus "inputs" in the FBP protocol, to communicate it from runtime to UI.

**Comment by forresto** [comment]
The biggest design challenge with this proposal is where to put the labels. 

I agree that visually denoting params would be good. Another concept:

![input and options](https://cloud.githubusercontent.com/assets/395307/5932328/42556464-a64f-11e4-829c-56a125f7347c.png)

https://github.com/the-grid/the-graph/issues/197

**Comment by ensonic** [comment]
Right, and if we treat inports as parameter ports as soon as one sets a static value, it might be confusing if the ports move around.

Can we do this for components that use wirepattern?

I agree with forresto that top and bottom ports wold be hard to label. Just moving stream ports up and having a gap to the other port would be nice already.

**Comment by jonnor** [comment]
I think the parameter ports should be indicated by port colo/shape rather than their position. 

Position influences heavily the routing of edges, and should be free to move around for this purpose. @forresto we used to move ports around based on the direction of their edges. Why did we stop doing that?

**Comment by ensonic** [comment]
@jonnor there are things like r/g/b or x/y/z where I think it is confusing if they swap their order.

**Comment by jonnor** [comment]
Fair point. For quick visual recognition (pattern matching) having the order the same is also beneficial.

**Comment by forresto** [comment]
I made that call for quick visual recognition when we switched from custom elements to svg/zui.

**Comment by forresto** [comment]
One thing to double-check is that ports should always be ordered as authored.
